shared/src/interfaces/stt_model.py

```diff
@@ -1,55 +1,142 @@
 
+import os
+import sys
 from abc import ABC, abstractmethod
-from typing import List, Dict, Any
+from typing import Any, Dict, List
+
+# Add the parent directory to the path for imports
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))
+
+from models import AudioRequest, ModelConfig, TranscriptionResult
 
 
 class SpeechToTextModel(ABC):
-    """Abstract base class for speech-to-text model implementations."""
-    
+    """Abstract base class for speech-to-text model implementations.
+
+    This interface defines the contract that all STT model implementations
+    must follow, ensuring consistent behavior across different model types.
+    """
+
+    def __init__(self, config: ModelConfig):
+        """Initialize the model with configuration.
+
+        Args:
+            config: Model configuration containing type, path, and parameters
+        """
+        self.config = config
+        self._is_loaded = False
+
     @abstractmethod
-    async def transcribe(self, audio_path: str, audio_format: str = None) -> str:
-        """Convert audio file to text.
-        
+    def transcribe(self, request: AudioRequest) -> TranscriptionResult:
+        """Transcribe audio to text.
+
         Args:
-            audio_path: Path to the audio file to transcribe
-            audio_format: Optional format hint for the audio file
-            
+            request: Audio request containing file path and processing options
+
         Returns:
-            Transcribed text from the audio file
-            
+            TranscriptionResult containing the transcribed text and metadata
+
         Raises:
-            STTError: If transcription fails
+            TranscriptionError: If transcription fails
+            AudioFormatError: If audio format is not supported
+            ModelLoadError: If model is not properly loaded
         """
         pass
-    
+
     @abstractmethod
     def get_supported_formats(self) -> List[str]:
-        """Return list of natively supported audio formats.
-        
+        """Get list of supported audio formats.
+
         Returns:
             List of supported audio format extensions (e.g., ['wav', 'mp3'])
         """
         pass
-    
+
     @abstractmethod
     def get_model_info(self) -> Dict[str, Any]:
-        """Return model metadata and capabilities.
-        
+        """Get information about the model.
+
         Returns:
-            Dictionary containing model information such as name, version, capabilities
+            Dictionary containing model information such as:
+            - name: Model name
+            - version: Model version
+            - type: Model type (whisper, mock, etc.)
+            - parameters: Current model parameters
+            - loaded: Whether model is loaded
         """
         pass
-    
+
     @abstractmethod
-    async def health_check(self) -> bool:
-        """Verify model is ready for processing.
-        
+    def health_check(self) -> Dict[str, Any]:
```

Comment:
ok, I'd like to hear justification to changes it from "async" to sync version. It might be worth it, but please explain why, or keep async, until we actually implement it